init.lua

```diff
@@ -2398,6 +2418,28 @@ if autocomplete.add_icon then
   end
 end
 
+local function get_current_symbol_info(doc)
+  local line1, col1, line2, col2 = doc:get_selection()
+  local symbol = doc:get_text(line1, col1, line2,col2)
+  if #symbol == 0 then
+    line1, col1, line2, col2 = get_token_range(doc, line1, col1)
+    if line1 ~= line2 then 
+      core.error("Impossible to get current symbol")
+      return nil, line1, col1, line2, col2
+    end
+    symbol = doc:get_text(line1, col1, line2, col2)
+    while #symbol > 0 and string.byte(symbol, 1) == 32 do
+      symbol = string.sub(symbol, 2)
+      col1 = col1 + 1
+    end
+    while #symbol > 0 and string.byte(symbol, #symbol) == 32 do
+      symbol = string.sub(symbol, 1, #symbol-1)
+      col2 = col2 -1
+    end
```

Comment:
You could simplify this a bit with something like:
```lua
space_start, symbol, space_end = string.match(symbol, "^(%s*)(.-)(%s*)$")
col1 = col1 + #space_start
col2 = col2 - #space_end
```

Reply:
My doubt in using regex is about their performances, are regex more performant than a simple while loop?

Reply:
This function is called once on a string, and even if this string is relatively big, it shouldn't have a big impact.
In some cases the pattern will also be faster, as we don't cross the Lua<->C barrier as much.

Also the pattern will take care of tabs and newlines too.